token/confidential-transfer/elgamal-registry/src/instruction.rs

```diff
@@ -0,0 +1,196 @@
+use {
+    crate::{get_elgamal_registry_address, id},
+    solana_program::{
+        instruction::{AccountMeta, Instruction},
+        program_error::ProgramError,
+        pubkey::Pubkey,
+        system_program, sysvar,
+    },
+    solana_zk_sdk::zk_elgamal_proof_program::{
+        instruction::ProofInstruction, proof_data::PubkeyValidityProofData,
+    },
+    spl_token_confidential_transfer_proof_extraction::{ProofData, ProofLocation},
+};
+
+#[derive(Clone, Debug, PartialEq)]
+#[repr(u8)]
+pub enum RegistryInstruction {
+    /// Initialize an ElGamal public key registry.
+    ///
+    /// 0. `[writeable, signer]` The funding account (must be a system account)
+    /// 1. `[writable]` The account to be created
+    /// 2. `[]` The wallet address (will also be the owner address for the
```

Comment:
Shouldn't this be a signer? Otherwise it's possible to create someone else's registry entry.

Side note: since this will require a signature from the owner, it means that SPL Token multisigs are not supported. This might be totally fine though

Reply:
Oh you are completely right. For some reason I thought the owner can always update the registry account with its own key, but obviously the owner may not want to have want a registry. Fixed!

Yeah I thought about the Token multisigs when I was implementing this, but I think this should be fine.